### ISSUE TITLE
[Edge Storage API] uploadFile function uploads the file as a folder

### ISSUE THREAD

**Issue by clementoriol** [jlarmstrongiv/bunny-client#2]
Hello ! 
Thank you for your great work on this client :) 
It's great to find a good, typescript-first client for the Bunny API.

While trying it, I stumbled on the following issue with the uploadFile function : 

**The issue :** 
Any attempt to upload a file using the following code : 
```
export const bunnyStorageClient = createManageFilesClient({
  storageZonePassword: process.env.BUNNY_STORAGE_ZONE_PASSWORD,
  storageZoneEndpoint: process.env.BUNNY_STORAGE_ZONE_ENDPOINT,
  storageZoneName: process.env.BUNNY_STORAGE_ZONE_NAME,
});

await bunnyStorageClient("uploadFile", {
    fileName: "image.jpg",
    file, // Buffer
  })
```

Will result in the creation of an empty folder named `image.jpg` in the storage space.

![Capture d’écran 2023-06-02 à 11 20 13](https://github.com/jlarmstrongiv/bunny-client/assets/3337108/e8445d27-cc08-485b-9130-1e3cce5bc456)

**Expected result**
The image file should be uploaded, and no extra folder created

**Suggested Fix**
I pinpointed the issue to this line : 

https://github.com/jlarmstrongiv/bunny-client/blob/12880c617803b5d68b6052e3ea9ca6c3689eff3c/src/edge-storage-api/manage-files/uploadFile/uploadFile.ts#L108

The extra trailing slash in the `overrideUrl` variable should be removed, as it instructs the Bunny API to create a folder instead of a file

**Comment by jlarmstrongiv** [comment]
Hi @clementoriol !  I’m glad you are finding the library useful :)

That’s a classic case of fixing one bug caused another.  When I added the delete file api, I realized it didn’t delete the nested folders—just the file.  Adding that extra slash fixed it.  

However, it seems that extra slash broke file uploading.  I think the best fix would be to detect whether the `fileName` has a trailing slash, and if so, append it to the `overrideUrl` (probably better than detecting file extensions).

I’ll fix this bug and publish a new version.  I’d recommend pinning the exact version.  While I took a break this month for my wedding, I plan to rapidly iterate on it again soon in the 0.x.x versions.

**Comment by clementoriol** [comment]
He @jlarmstrongiv thanks for the answer 🥇 
Don't worry too much about it, that's the life of a new library 😄

Following your message I also tested the `deleteFile` fn and, yeah,  got the same issue, the trailing slash prevents file deletion there too.

I'm not sure I understand your use-case about the nested folder deletions, do you have any example ? 
From my perspective, since the functions expects a (required) `fileName` parameter I'd suppose they only delete the passed file. Maybe there should be a different function or option for folder deletion ? The `fileName` with a trailing slash thing seems a bit confusing.

Anyway, hope this helps, and don't hesitate to ask if I can submit a PR or anything

And congrats on your wedding 🎉

**Comment by jlarmstrongiv** [comment]
Hey @clementoriol 😄

> I'm not sure I understand your use-case about the nested folder deletions, do you have any example ?

The issue occurs when creating `/path/to/file.txt`.  It’s like `mkdir -p`, which recursively creates folders if they do not already exist.  When deleting that file, it leaves the folder `/path/to/`.  

I realized that the browser-compatible normalize path package I am using strips trailing slashes by default.  Adding back the trailing slash manually let me delete folders, but broke file operations.  

I have fixed the bug in 0.0.7, by removing my manual trailing slash and disabling the `stripTrailing` from the normalize path package.  That way, the users’ intent of referencing a file or folder is now preserved.  

> Maybe there should be a different function or option for folder deletion ?

I agree that it’s a bit confusing.  In comparison, the s3 api has no concept of folders, and just has `/` in the file name itself—it’s the UI that treats the `/` in the file name as folders.

While I would have made different API decisions in some cases, I’m going to try and preserve most of the API compatibility.

> Anyway, hope this helps, and don't hesitate to ask if I can submit a PR or anything

If you can let me know whether it works or not, that’d be great!  If not, please let me know and I’ll re-open this issue.

> And congrats on your wedding 🎉

Thank you 🎉 it’s been wonderful

**Comment by clementoriol** [comment]
Hey @jlarmstrongiv, amazing ! 

It works great from what I've tested : 
- ✅ file upload
- ✅ file deletion
- ✅ folder deletion

Thank you ! 🥳
